x/shared/session.go

```diff
@@ -106,3 +107,45 @@ func GetProofWindowCloseHeight(sharedParams *sharedtypes.Params, queryHeight int
 	return GetProofWindowOpenHeight(sharedParams, queryHeight) +
 		int64(sharedParams.GetProofWindowCloseOffsetBlocks())
 }
+
+// GetEarliestClaimCommitHeight returns the earliest block height at which a claim
+// for the session that includes queryHeight can be committed for a given supplier
+// and the passed sharedParams.
+func GetEarliestClaimCommitHeight(
+	sharedParams *sharedtypes.Params,
+	queryHeight int64,
+	claimWindowOpenBlockHash []byte,
```

Comment:
Would it not be easier to just give the `queryHeight` and get `claimWindowOpenBlockHash` internally given that we already have access to `sharedParams`?

Reply:
This static function has no way to query for the claim window open block such that it can get a reference to its hash.